oak-core/src/test/java/org/apache/jackrabbit/oak/query/AbstractQueryTest.java

```diff
@@ -312,10 +316,27 @@ protected void assertResultSize(String query, String language, long expected) {
 
     protected List<String> assertQuery(String sql, String language,
                                        List<String> expected, boolean skipSort) {
+        return assertQuery(sql, language, expected, skipSort, false);
+    }
+
+    protected List<String> assertQuery(String sql, String language,
+                                       List<String> expected, boolean skipSort, boolean checkSort) {
         List<String> paths = executeQuery(sql, language, true, skipSort);
-        assertResult(expected, paths);
+        if (checkSort) {
+            assertResult_Sorted(expected, paths);
+        } else {
+            assertResult(expected, paths);
+        }
         return paths;
+    }
 
+    protected static void assertResult_Sorted(@NotNull List<String> expected, @NotNull List<String> actual) {
+        assertEquals("Result set size is different: " + actual, expected.size(),
+                actual.size());
+
+        for (int i = 0; i < expected.size(); i++) {
+            assertEquals("Expected sorted result not found", expected.get(i), actual.get(i));
+        }
```

Comment:
can we avoid checking every single entry and just do an `assertEquals` on the lists?